### PR TITLE
[gatsby-plugin-sharp] Fix PNG generation when using the "duotone" option

### PR DESCRIPTION
**Pull request by fk**
Options for PNG and JPEG output need to be set prior to `raw().toBuffer()` so that `toFormat()` in duotone.js actually works.

Things were working for JPEGs before because they are written `toFile()` as JPEGs explicitly. PNGs get processed with imagemin before writing them to disk, which requires a valid (PNG) buffer.

When authoring the original feature, I moved the PNG/JPEG output settings after the actual duotone processing (for whatever reason 😬  – I guess I thought they get dropped when doing `raw().toBuffer()`) – this PR reverts that, and cleans up duotone.js.

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit 5f9ae44a8b93da18d8bf749b06a399dc5980383d

https://app.netlify.com/sites/using-styled-components/deploys/5968f5d4cf321c72b7f87157

File: packages/gatsby-plugin-sharp/src/duotone.js [hunk 1]
```diff
@@ -1,45 +1,44 @@
...
+    hexToRgb(duotone.highlight),
+    hexToRgb(duotone.shadow)
+  )
 
-    if (format === `jpg`) {
-      format = `jpeg`
-    }
+  // @todo remove once we upgrade to sharp v0.18 which
+  // adds an alias for "jpg" to toFormat
+  // @see http://sharp.dimens.io/en/stable/changelog/#v0180-30th-may-2017
```
Comment (KyleAMathews) [inline packages/gatsby-plugin-sharp/src/duotone.js]: This seems like the main blocker for upgrading as we "attention" is the default mode for us for cropping https://github.com/jcupitt/libvips/issues/672#issuecomment-307365416

**Comment by KyleAMathews** [comment]
Everything looks happy 👍 https://596ebdce8ebdd96db650f3e0--image-processing.netlify.com/

**Comment by fk** [comment]
👍  @KyleAMathews I actually should have added a PNG-and-duotone example to the image-processing example … :-/. Will do! I wish I could get my head around writing tests for this.

File: packages/gatsby-plugin-sharp/src/duotone.js (hunk 1 above)
Comment (fk) [inline packages/gatsby-plugin-sharp/src/duotone.js]: Thanks for the heads up! Didn't have the time to look into the changelog for 0.18 yet, merely stumbled upon that note regarding the alias. "attention" is all I had to mention to the people that I showed plugin-sharp to instantly win them over btw. <3

**Pull request merged**